tests/cupy_tests/manipulation_tests/test_basic.py

```diff
@@ -125,3 +126,44 @@ def test_copyto_where(self, xp, dtype):
             self.dst_shape, xp, dtype) % 2).astype(xp.bool_)
         xp.copyto(dst, self.src, where=mask)
         return dst
+
+
+@testing.parameterize(
+    *testing.product(
+        {'shape': [(0,), (1,), (2, 3), (2, 3, 4)]}))
+@testing.gpu
+class TestPutmask(unittest.TestCase):
+
+    @testing.for_all_dtypes(no_complex=True, no_bool=True)
```

Comment:
```suggestion
    @testing.for_all_dtypes()
```

Reply:
The operation(power) used for the array `values` in this test can not be performed on `bool`s. That's why I am going to write a separate test for `bool`s.

Reply:
So, do not use power and modulo in this test.